src/executor/analyze_executor.cpp

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// analyze_executor.cpp
+//
+// Identification: src/executor/analyze_executor.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <vector>
+
+#include "executor/analyze_executor.h"
+#include "executor/executor_context.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "common/logger.h"
+#include "catalog/catalog.h"
+
+namespace peloton {
+namespace executor {
+
+AnalyzeExecutor::AnalyzeExecutor(const planner::AbstractPlan *node,
+                                 ExecutorContext *executor_context)
+    : AbstractExecutor(node, executor_context),
+      executor_context_(executor_context) {}
+
+bool AnalyzeExecutor::DInit() {
+  LOG_TRACE("Initializing analyze executor...");
+  LOG_TRACE("Analyze executor initialized!");
+  return true;
+}
+
+bool AnalyzeExecutor::DExecute() {
+  LOG_TRACE("Executing Analyze...");
+
+  const planner::AnalyzePlan &node = GetPlanNode<planner::AnalyzePlan>();
+
+  UNUSED_ATTRIBUTE storage::DataTable* target_table = node.GetTable();
+  UNUSED_ATTRIBUTE std::vector<char*> target_columns = node.GetColumnNames();
+
+  LOG_TRACE("Analyzing table %s", node.GetTableName().c_str());
+  LOG_TRACE("Analyzing column size %lu", target_columns.size());
+
+  auto current_txn = executor_context_->GetTransaction();
+  current_txn->SetResult(peloton::ResultType::SUCCESS);
+
+  // auto &transaction_manager =
+  //     concurrency::TransactionManagerFactory::GetInstance();
+
+
+  // stats_storage here
```

Comment:
@wangxiaoyou11 Here is the entry point for stats collection. Let me know if you need more information.